session/Concept.proto

```diff
@@ -151,56 +122,85 @@ message Method {
             // Thing method responses
             Thing.Type.Res thing_type_res = 900;
             Thing.IsInferred.Res thing_isInferred_res = 901;
-            Thing.Keys.Iter thing_keys_iter = 902;
-            Thing.Attributes.Iter thing_attributes_iter = 903;
-            Thing.Relations.Iter thing_relations_iter = 904;
-            Thing.Roles.Iter thing_roles_iter = 905;
             Thing.Relhas.Res thing_relhas_res = 906;
             Thing.Unhas.Res thing_unhas_res = 907;
 
             // Relation method responses
-            Relation.RolePlayersMap.Iter relation_rolePlayersMap_iter = 1000;
-            Relation.RolePlayers.Iter relation_rolePlayers_iter = 1001;
             Relation.Assign.Res relation_assign_res = 1002;
             Relation.Unassign.Res relation_unassign_res = 1003;
 
             // Attribute method responses
             Attribute.Value.Res attribute_value_res = 1100;
-            Attribute.Owners.Iter attribute_owners_iter = 1101;
         }
     }
 
     message Iter {
-        message Res {
-            // SchemaConcept iterator responses
-            SchemaConcept.Sups.Iter.Res schemaConcept_sups_iter_res = 205;
-            SchemaConcept.Subs.Iter.Res schemaConcept_subs_iter_res = 206;
-
-            // Role iterator responses
-            Role.Relations.Iter.Res role_relations_iter_res = 401;
-            Role.Players.Iter.Res role_players_iter_res = 402;
-
-            // Type iterator responses
-            Type.Instances.Iter.Res type_instances_iter_res = 502;
-            Type.Keys.Iter.Res type_keys_iter_res = 503;
-            Type.Attributes.Iter.Res type_attributes_iter_res = 504;
-            Type.Playing.Iter.Res type_playing_iter_res = 505;
-
-            // RelationType iterator responses
-            RelationType.Roles.Iter.Res relationType_roles_iter_res = 701;
-
-            // Thing iterator responses
-            Thing.Keys.Iter.Res thing_keys_iter_res = 902;
-            Thing.Attributes.Iter.Res thing_attributes_iter_res = 903;
-            Thing.Relations.Iter.Res thing_relations_iter_res = 904;
-            Thing.Roles.Iter.Res thing_roles_iter_res = 905;
-
-            // Relation iterator responses
-            Relation.RolePlayersMap.Iter.Res relation_rolePlayersMap_iter_res = 1000;
-            Relation.RolePlayers.Iter.Res relation_rolePlayers_iter_res = 1001;
+        message Req {
+            oneof iter_req {
```

Comment:
let's rename `iter_req` to `req` to be consistent with other `oneof` variable names (being just the lowercasing of their owning `message`).

Reply:
The rationale here is that this is in an `Iter.Req` message, and all of the children of `Iter.Res` are named: `xxx_xxx_iter_req`, so I think this way is actually more consistent.

Reply:
Right, I can see that. But in this case, we've been using just the name of parent on-level up, rather than two-level up. For example, `oneof answer`. So let's keep it just `req` to be consistent.